doc/control_mechanisms/quest_callbacks.md

```diff
@@ -0,0 +1,74 @@
+---
+title: "Konzept: Quest Callbacks"
+---
+
+## Ziel
+
+Mithilfe von Callbacks können in der DSL Methoden definiert werden, die beim Auftreten bestimmter Events im Spiel ausgelöst werden. 
+Diese Event-Handler-Methoden sollen genutzt werden, um z.B. die vom Spieler abgegebene Lösung für eine Quest aus dem Spiel zurück an die DSL zu übergeben, um dort die Lösung zu bewerten und ggf. über native Funktionen im Dungeon spielerische Konsequenzen auszulösen.
+
+## Von der Quest zur Entität und wieder zurück 
+
+*Anmerkung:* Das Konzept ist noch WiP und hier wird lediglich der aktuelle Stand festgehalten. 
+
+![Von der Quest zur Entität und wieder zurück.](img/quest.png)
+
+Das Diagramm zeigt, wie eine Quest im Spiel implementiert ist und wie diese mit der Entität verbunden wird. 
+
+Jeder Aufgabentyp im Spiel wird als abstrakte `Quest` definiert. Von dort werden die konkreten Questtypen wie `QuizQuestionQuest` oder `ReplacementQuest` abgeleitet. 
+
+Jede Quest speichert eine Menge an `QuestContent`-Objekten ab. Das können z.B. die Fragestellung oder die Antwortmöglichkeiten für eine Quizfrage sein.
+
+Eine in der DSL definierte Aufgabenstellung erzeugt beim Laden ein entsprechendes `Quest`-Objekt mit zugehörigen `QuestContent`-Objekten.
+
+Aus den jeweiligen `QuestContent`-Objekten werden dann `TaskReferenceComponent`s erstellt und an Entitäten im Dungeon angehängt. Ob jedes Content-Objekt ein eigenes Component ist oder auch mehrere Content-Objekte in einem Component zusammengefasst werden können, ist noch nicht ganz klar. 
```

Comment:
Wieso ein extra Objekt, warum nicht die Quest an die Entität knüpfen?

Reply:
In #819 hab ich den Eindruck bekommen: Beides.

Die Entität wird mit dem `TaskReferenceComponent` der `Quest` zugeordnet. 
Zeitgleich stellt eine Entität aber auch "nur" einen (oder mehrere?) Komponenten einer Quest dar. Z.B könnte ist ein Item die Antwortmöglichkeit B. Die Verbindung muss irgendwie gezogen werden.